crates/sui-core/src/checkpoints/mod.rs

```diff
@@ -870,7 +873,12 @@ impl CheckpointAggregator {
                 continue;
             }
 
-            match select(self.exit.changed().boxed(), self.notify.notified().boxed()).await {
+            match select(
```

Comment:
Flagging this change -- seems related to investigating the checkpoint bug, but I couldn't tell straight away whether it's destined to stay or not.